llvm/lib/Target/Xtensa/XtensaISelLowering.cpp

```diff
@@ -519,6 +537,87 @@ SDValue XtensaTargetLowering::LowerImmediate(SDValue Op,
   return Op;
 }
 
+SDValue XtensaTargetLowering::LowerGlobalAddress(SDValue Op,
+                                                 SelectionDAG &DAG) const {
+  const GlobalAddressSDNode *G = cast<GlobalAddressSDNode>(Op);
+  SDLoc DL(Op);
+  auto PtrVt = getPointerTy(DAG.getDataLayout());
```

Comment:
Use the type of the incoming value instead of guessing it's an address space 0 pointer. You should hardly ever need getPointerTy

Reply:
fixed